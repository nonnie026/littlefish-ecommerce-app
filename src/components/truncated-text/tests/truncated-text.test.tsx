import '@testing-library/jest-dom'
import { cleanup, render, screen } from '@testing-library/react'

import { TruncatedText } from '../index'

describe('TruncatedText', () => {
  const text = 'This is a test text for truncation'
  const minStringLength = 10

  afterEach(cleanup)

  const view = () =>
    render(<TruncatedText minStringLength={minStringLength} text={text} />)

  it('matches snapshot', () => {
    const { asFragment } = view()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the truncated text correctly', () => {
    view()
    const expectedTruncatedText = `${text.substring(0, minStringLength)}...`

    expect(screen.getByTestId('truncated-text')).toHaveTextContent(
      expectedTruncatedText
    )
  })

  it('displays the full text when shorter than minStringLength', () => {
    const shortText = 'Short'
    render(<TruncatedText minStringLength={minStringLength} text={shortText} />)

    expect(screen.getByTestId('truncated-text')).toHaveTextContent(shortText)
  })
})
