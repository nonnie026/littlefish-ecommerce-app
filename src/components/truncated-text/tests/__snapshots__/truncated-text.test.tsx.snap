// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncatedText matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root MuiTypography-body14 css-37jdci-MuiTypography-root"
    data-testid="truncated-text"
  >
    This is a ...
  </span>
</DocumentFragment>
`;
